### DISCUSSION TITLE
Getting a broken model on import.

### DISCUSSION THREAD

**Discussion by derik-dot-digital** [Sandman13sq/DmrVBM-blender-to-gms2#9] [Q&A] [answered]
[DmrVMB help.zip](https://github.com/Sandman13sq/DmrVBM-blender-to-gms2/files/14277468/DmrVMB.help.zip)
![Capture](https://github.com/Sandman13sq/DmrVBM-blender-to-gms2/assets/126590189/ac675c0f-8705-4484-9ff3-6375896a5f32)

this is how I'm importing the exported vbuffer from that project 
vbuff_melon = OpenVertexBuffer("assets/melon.vb", global.v_format, true);

and i'm trying to draw it like this:

matrix_set(matrix_world, matrix_build(room_width/2, room_height/2, 0, 0, 0, 0, 1000, 1000, 1000));
vertex_submit(vbuff_melon, pr_trianglelist, -1);

no idea why the result looks so corrupted, in the attached file is my blender project for the object

**Comment by Sandman13sq** [comment, marked as answer]
The addon had some attribute building errors where normals used the wrong array size and colors would be missing if no vertex color layers were found for a mesh. This has been fixed in the [latest commit](https://github.com/Sandman13sq/DmrVBM-blender-to-gms2.git).

Your format defined in the Blender file you sent uses color floats. If this is intended, the color attribute will differ GM and will show a stride error when not set correctly.

`vertex_format_add_color()` is native GM, but will give a stride error with your current vb format:
![GameMaker-LTS_J6D17dOxPp](https://github.com/Sandman13sq/DmrVBM-blender-to-gms2/assets/55517515/519daaef-d35b-4826-b72d-551b26e0c94a)

`vertex_format_add_custom(vertex_type_float4, vertex_usage_color)` matches what you have defined in your Blender file:
![Runner_kLpZ96aDzz](https://github.com/Sandman13sq/DmrVBM-blender-to-gms2/assets/55517515/7f76f4df-ffbe-4abd-a0f2-851bee13e1d4)

You can change the format either in GM or Blender to match the other. I suggest `Color Bytes` since it saves on space at the cost of very small accuracy loss.
(idk if you already have this part sorted out. Just some extra context.)

**Comment by derik-dot-digital** [comment]
Thanks so much for the quick response! And also thanks for the tip on the color usage, I had noticed I had it wrong in my testing. Awesome work man!
